airflow/providers/elasticsearch/log/es_task_handler.py

```diff
@@ -70,17 +70,6 @@ def get_es_kwargs_from_config() -> dict[str, Any]:
         if elastic_search_config
         else {}
     )
-    # For elasticsearch>8 retry_timeout have changed for elasticsearch to retry_on_timeout
-    # in Elasticsearch() compared to previous versions.
-    # Read more at: https://elasticsearch-py.readthedocs.io/en/v8.8.2/api.html#module-elasticsearch
-    if (
-        elastic_search_config
-        and "retry_timeout" in elastic_search_config
-        and not kwargs_dict.get("retry_on_timeout")
-    ):
-        retry_timeout = elastic_search_config.get("retry_timeout")
-        if retry_timeout is not None:
-            kwargs_dict["retry_on_timeout"] = retry_timeout
```

Comment:
I think this for backcompat with previous config.

The problem here that we never raise Depreciation Warning here and just silently replace `retry_timeout` to `retry_on_timeout`
In theory it could be breaking changes in the other hand note to 5.0.0 release could cover this change https://airflow.apache.org/docs/apache-airflow-providers-elasticsearch/stable/changelog.html#id9